src/Helper/ServerUrl.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Mvc\View\Helper;
+
+use function assert;
+use function is_string;
+use function ltrim;
+use function rtrim;
+
+final class ServerUrl
```

Comment:
This view helper will be used in a laminas-mvc based application instead of the view helper in laminas-view. And the `ServerUrl` of laminas-view remains and is not touched.
Right?

https://github.com/laminas/laminas-view/blob/77a4b6d78445ae2f30625c5af09a05ad4e4434eb/src/Helper/ServerUrl.php#L19

Reply:
> In view 3, we simply delete the old helpers…

Why? The ServerUrl helper uses `$_SERVER` therefore this helper is independent of laminas-http or laminas-router or laminas-mvc.

Reply:
I guess it _is_ independent from these deps. I'm not sure why we'd want to keep it in view after 3 though. In a Mezzio app, the server url helper is configured in middleware using the request object, in MVC, $_SERVER is consulted… on the cli - we've had it. If anything, the ServerUrl helper in view should be replaced with something very simple that just accepts a configured url to its constructor - similar to the plans laid out in https://github.com/laminas/laminas-view/pull/160 ??

Reply:
Sorry - I wasn't looking at this in context - what I've done here would work fine on the cli because it prefers a configured url over the one detected by the environment.

Reply:
> [Stand-alone?](https://docs.laminas.dev/laminas-view/application-integration/stand-alone/)

In `view`, I think we should replace the current impl. with a simple factory driven helper that _requires_ a pre-configured url, i.e. `['view_manager' => ['server_url' => 'https://example.com']]`

Reply:
> …simple factory driven helper…

Perfect! 👍🏻